packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts

```diff
@@ -0,0 +1,126 @@
+import { NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { DataSource, Repository } from 'typeorm';
+
+import { decryptText } from 'src/engine/core-modules/auth/auth.util';
+import {
+  RemoteServerType,
+  RemoteServerEntity,
+} from 'src/engine/metadata-modules/remote-server/remote-server.entity';
+import { RemoteTableStatus } from 'src/engine/metadata-modules/remote-server/remote-table/dtos/remote-table.dto';
+import { EnvironmentService } from 'src/engine/integrations/environment/environment.service';
+import { WorkspaceDataSourceService } from 'src/engine/workspace-datasource/workspace-datasource.service';
+import { EXCLUDED_POSTGRES_SCHEMAS } from 'src/engine/metadata-modules/remote-server/remote-table/utils/excluded-postgres-schemas';
+
+export class RemoteTableService {
+  constructor(
+    @InjectRepository(RemoteServerEntity, 'metadata')
+    private readonly remoteServerRepository: Repository<
+      RemoteServerEntity<RemoteServerType>
+    >,
+    private readonly workspaceDataSourceService: WorkspaceDataSourceService,
+    private readonly environmentService: EnvironmentService,
+  ) {}
+
+  public async findAvailableRemoteTablesByServerId(
+    id: string,
+    workspaceId: string,
+  ) {
+    const remoteServer = await this.remoteServerRepository.findOne({
+      where: {
+        id,
+        workspaceId,
+      },
+    });
+
+    if (!remoteServer) {
+      throw new NotFoundException('Remote server does not exist');
+    }
+
+    switch (remoteServer.foreignDataWrapperType) {
+      case RemoteServerType.POSTGRES_FDW:
+        return this.findAvailableRemotePostgresTables(
+          workspaceId,
+          remoteServer,
+        );
+      default:
+        throw new Error('Unsupported foreign data wrapper type');
+    }
+  }
+
+  // TODO: may be moved into a separated postgres table service once we have more use cases
+  private async findAvailableRemotePostgresTables(
+    workspaceId: string,
+    remoteServer: RemoteServerEntity<RemoteServerType>,
+  ) {
+    const remotePostgresTables =
+      await this.fetchTablesFromRemotePostgresSchema(remoteServer);
+
+    const workspaceDataSource =
+      await this.workspaceDataSourceService.connectToWorkspaceDataSource(
+        workspaceId,
+      );
+
+    const currentForeignTableNames = (
```

Comment:
we should trust metadata here.
we should have remote object (with isRemote: true)